### PR TITLE
Separate carbon cycle

### PR DESCRIPTION
**Pull request by chrisroadmap**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [ ] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

Does not require example as does not affect underlying model operation or interface

### CODE REVIEW

**Comment by chrisroadmap** [comment]
Some of the parameters passed to `carbon_cycle` are the same as in `fair_scm` - there are about 15 arguments which is a little unwieldy as it stands and perhaps they can just be made global. On the other hand the benefit of including everything is that `carbon_cycle` can be run stand-alone.

**Comment by znicholls** [comment]
> perhaps they can just be made global

don't do that, global arguments will make your life a nightmare

> a little unwieldy as it stands

I think it's going to have to be that way for a little bit so you can make it easy to then later refactor into objects. Think of it as the storm before the calm and trust the tests to ensure that nothing fatal changes.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -3,6 +3,8 @@ Changelog
 
 master
 ------
+(`#60 <https://github.com/OMS-NetZero/FAIR/pull/60>`_) Refactoring: take carbon cycle out
```
Comment (znicholls) [inline CHANGELOG.rst]:
```suggestion
(`#60 <https://github.com/OMS-NetZero/FAIR/pull/60>`_) Refactor carbon cycle into separate function
```

File: fair/forward.py [hunk 2]
```diff
@@ -28,7 +28,109 @@ def iirf_interp_funct(alp_b,a,tau,iirf_h,targ_iirf):
...
+    """Simple linear iIRF relationship. Eq. (8) of Millar et al ACP (2017).
+    
+    Inputs:
+        c_acc    : cumulative airborne carbon anomaly (GtC) since
+                   pre-industrial
+        temp     : temperature anomaly since pre-industrial
+        r0       : pre-industrial time-integrated airborne fraction (yr)
+        rc       : sensitivity of time-integrated airborne fraction to airborne
+                   carbon (yr/GtC)
+        rt       : sensitivity of time-integrated airborne fraction to
```
Comment (znicholls) [inline fair/forward.py:42]: docstring for iirf_max?

**Review by znicholls — COMMENTED** [review]
nice

File: fair/forward.py (hunk 2 above)
Comment (chrisroadmap) [reply fair/forward.py:42]: Well-spotted. New commit includes this.

I wonder if docstringing every single function is overkill. Probably only the ones that are intended to be external are necessary.

Comment (znicholls) [reply fair/forward.py:42]: Good practice is to add docstrings to all public functions. You have a lot of public functions so you might 'mark' a few more as private and then docstrings are more optional/for developer use (I say 'mark' as private as there's no such thing as private in Python really). 

As part of 2.0 refactoring let's talk more about some ways to tidy that sort of stuff up. For now it's just unnecessary extraneous noise so don't stress.

**Pull request merged**